### PR TITLE
Fix tsx workaround, bootstrap next.d.ts, add tslint deps

### PR DESCRIPTION
**Pull request by brikou**
The primary goal of this PR is to solve issue explain in README by setting jsx to react-native.

Another big idea is to bootstrap a next.d.ts definition file as a reference (quite empty for now) but this is a bootstrap.

I also revamp the example to use next/link and so emphasize that there is no more problem :)

### CODE REVIEW

File: examples/with-typescript/tslint.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "defaultSeverity": "error",
+  "extends": ["tslint:recommended"],
+  "jsRules": {},
+  "rules": {
+    "quotemark": [true, "single", "jsx-double"],
+    "semicolon": [true, "never"]
```
Comment (brikou) [inline examples/with-typescript/tslint.json]: I've changed default rules, to match next.js coding style (but I don't like it). Is it possible to stick to default rules (using double quote and always append semicolons).

File: examples/with-typescript/package.json [hunk 2]
```diff
@@ -2,16 +2,21 @@
...
-    "react": "^15.5.4",
-    "react-dom": "^15.5.4"
+    "react": "^15.6.1",
+    "react-dom": "^15.6.1"
   },
   "devDependencies": {
-    "@types/react": "^15.0.1",
-    "concurrently": "^3.1.0",
-    "typescript": "^2.1.5"
+    "@types/node": "8.0.26",
```
Comment (brikou) [inline examples/with-typescript/package.json]: This definitions are required by next.d.ts (and url import)

**Comment by brikou** [comment]
@timneutkens do you want me to rebase my PR, otherwise just close it, no problem

**Comment by timneutkens** [comment]
@brikou looks okay, just rebase and ping me again, I'll merge 👌

**Comment by brikou** [comment]
@timneutkens rebased :+1:

**Review by timneutkens — APPROVED** [review]
Thanks!

**Comment by brikou** [comment]
@timneutkens thank you too!

**Pull request merged**